File: article/src/main/java/com/ruzhan/awaker/article/network/HttpClient.java
package com.ruzhan.awaker.article.network;

import okhttp3.OkHttpClient;

public class HttpClient {

    private static HttpClient httpClient;

    private OkHttpClient okHttpClient;

    public static HttpClient get() {
        if (httpClient == null) {
            synchronized (HttpClient.class) {
                if (httpClient == null) {
                    httpClient = new HttpClient();
                }
            }
        }
        return httpClient;
    }

    private HttpClient() {
        okHttpClient = new OkHttpClient.Builder().build();
    }

    public static OkHttpClient getHttpClient() {
        return get().okHttpClient;
    }
}

File: article/src/main/java/com/ruzhan/awaker/article/network/HttpClient.kt
package com.ruzhan.awaker.article.network

import okhttp3.OkHttpClient

class HttpClient private constructor() {

    private val okHttpClient: OkHttpClient

    init {
        okHttpClient = OkHttpClient.Builder().build()
    }

    companion object {

        private var httpClient: HttpClient? = null

        fun get(): HttpClient {
            if (httpClient == null) {
                synchronized(HttpClient::class.java) {
                    if (httpClient == null) {
                        httpClient = HttpClient()
                    }
                }
            }
            return httpClient!!
        }

        fun getHttpClient(): OkHttpClient {
            return get().okHttpClient
        }
    }
}
